azure.pkr.hcl

```diff
@@ -103,7 +111,7 @@ source "azure-arm" "spacelift" {
       resource_group       = var.gallery_resource_group
       gallery_name         = var.gallery_name
       image_name           = var.gallery_image_name
-      image_version        = var.gallery_image_version
+      image_version        = "${var.gallery_image_version}-${local.image_architecture}"
```

Comment:
I don't think this approach will work because I don't think an Azure compute gallery image supports publishing multiple architectures / VM generations to the same image. I think what we might need to do is create a new image in the compute gallery (it's configured [here](https://github.com/spacelift-io/infra-azure/blob/bd1a3e25158f1df951cf2b2e1964829d4317d72b/public/worker_images/images.tf#L26) and managed by [this](https://spacelift-io.app.spacelift.io/stack/azure-public) stack).

Give me a shout if you need a hand with this.